aiu_fms_testing_utils/utils/decoders.py

```diff
@@ -0,0 +1,333 @@
+                )
+                prompt2 = template.format("Explain some popular greetings in Spanish.")
+                prompt3 = template.format("Explain to me why ignorance is bliss.")
+                prompt4 = template.format(
+                    "I have just come into a very large sum of money. Provide me a list of things that I can do with my new found wealth."
+                )
+            elif args.prompt_type == "code":
+                template = "[INST] Write code to solve the following coding problem that obeys the constraints and passes the example test cases. Please wrap your code answer using ```:\n{}\n[/INST]"
+                prompt1 = template.format("Write a bubble sort function in python.")
+                prompt2 = template.format(
+                    "Using the Java streams API, write a simple function which will get the cumulative sum of a list of integers."
+                )
+                prompt3 = template.format(
+                    "In bash, how do I list all directories and sub-directories which contain a .py file."
+                )
+                prompt4 = template.format(
+                    "Write a simple decorator in python which will modify all string inputs to ints if possible."
+                )
+            else:
+                dprint("prompt_type must be one of chat or code")
+                exit()
+
+            prompt1 = self.ids_for_prompt(prompt1)
+            prompt2 = self.ids_for_prompt(prompt2)
+            prompt3 = self.ids_for_prompt(prompt3)
+            prompt4 = self.ids_for_prompt(prompt4)
+            prompts = [prompt1, prompt2, prompt3, prompt4]
+            prompts = prompts * ((args.batch_size // 4) + 1)
+            prompts = prompts[: args.batch_size]
+
+        if args.fixed_prompt_length != 0:
+            padding_length = args.fixed_prompt_length
+            max_allowed_length = args.fixed_prompt_length
+        else:
+            padding_length = args.min_pad_length
+            max_allowed_length = args.max_prompt_length
+
+        self.has_padding = args.batch_size > 1 or padding_length != 0
+        self.max_len = max([len(prompt) for prompt in prompts])
+
+        if args.fixed_prompt_length != 0 and args.fixed_prompt_length < self.max_len:
+            dprint(
+                "One or more prompts require truncation. Truncation has been disabled "
+                "because fixed_prompt_length was set."
+            )
+            sys.exit(1)
+        prompts = self.truncate_prompts_to_max_length(
+            prompts,
+            self.max_len,
+            max_allowed_length,
+        )
+        if self.has_padding:
+            ids, extra_generation_kwargs = pad_input_ids(
+                prompts,
+                min_pad_length=padding_length,
+            )
+        else:
+            ids = prompts
+            if isinstance(ids, list) and len(ids) == 1:
+                ids = ids[0].unsqueeze(0)
+            extra_generation_kwargs = None
```

Comment:
I think it's better to have this be an empty dict instead with the new attention_kwargs work